src/neuralnet/cpid.h

```diff
@@ -344,4 +344,36 @@ class MiningId
     //!
     boost::variant<Invalid, Investor, Cpid> m_variant;
 }; // MiningId
+} // namespace NN
+
+namespace std {
+//!
+//! \brief Specializes std::hash<T> for NN::Cpid.
+//!
+//! This enables the use of NN::Cpid as a key in a std::unordered_map object.
+//!
+//! CONSENSUS: Don't use the hash produced by this routine (or by any std::hash
+//! specialization) in protocol-specific implementations. It ignores endianness
+//! and outputs a value with a chance of collision probably too great for usage
+//! besides the intended local look-up functionality.
+//!
+template<>
+struct hash<NN::Cpid>
+{
+    //!
+    //! \brief Create a hash of the supplied CPID object.
+    //!
+    //! \param cpid Contains the bytes to hash.
+    //!
+    //! \return A hash as the sum of the two halves of the bytes in the CPID.
+    //!
+    size_t operator()(const NN::Cpid& cpid) const
```

Comment:
`size_t` is platform dependent (32 or 64 bit). Will that matter here?

Reply:
I don't think it matters how much it wraps...from what I understand, `unordered_map` does an equality check after the hash lookup.